packages/template-retail-react-app/app/pages/login/index.jsx

```diff
@@ -170,8 +170,10 @@ const Login = ({initialView = LOGIN_VIEW}) => {
     useEffect(() => {
         if (isRegistered) {
             handleMergeBasket()
-            if (location?.state?.directedFrom) {
-                navigate(location.state.directedFrom)
+            const locatedFrom = window.localStorage.getItem('returnToPage')
+            window.localStorage.removeItem('returnToPage')
+            if (locatedFrom) {
```

Comment:
We should call removeItem if it is available. 
```suggestion
            if (locatedFrom) {
               window.localStorage.removeItem('returnToPage')
            }
```